docs/core/configuration.md

```diff
@@ -604,3 +604,54 @@ Here's a brief summary of the timeouts:
 - `timeout_commit` = how long a validator should wait after committing a block, before starting
   on the new height (this gives us a chance to receive some more precommits,
   even though we already have +2/3)
+
+### The effect of `timeout_propose` on the proposer selection process
```

Comment:
this is good, I think eventually we might also have to explain other timeouts (prevote, precommit, commit) and how they related to each other.